src/System.Runtime.Extensions/src/System/Collections/Hashtable.cs

```diff
@@ -153,6 +153,9 @@ private struct bucket
         private IEqualityComparer _keycomparer;
         private Object _syncRoot;
 
+        private static ConditionalWeakTable<object, SerializationInfo> s_serializationInfoTable;
+        private static ConditionalWeakTable<object, SerializationInfo> SerializationInfoTable => LazyInitializer.EnsureInitialized(ref s_serializationInfoTable);
```

Comment:
In CoreLib we use Interop.CompareExchange instead. @jkotas any preference here?

```csharp
internal static ConditionalWeakTable<object, SerializationInfo> SerializationInfoTable
        {
            get
            {
                if (s_serializationInfoTable == null)
                    Interlocked.CompareExchange(ref s_serializationInfoTable, new ConditionalWeakTable<object, SerializationInfo>(), null);

                return s_serializationInfoTable;
            }
        }
```

Reply:
I do not have preference.

`LazyInitializer.EnsureInitialized` is convenience helper. It looks nicer, but it results into bigger slower code.

Reply:
thanks :) I'm fine with using LazyInitializer here as the serialization code path isn't highly perf related.